plugins/policy-recommendation/policy_recommendation_job.py

```diff
@@ -0,0 +1,843 @@
+                egress = egressRules,
+            )
+        )
+    return [dict_to_yaml(np.to_dict())]
+
+def generate_reject_acnp(applied_to):
+    if not applied_to:
+        np_name = "recommend-reject-all-acnp"
+        applied_to = antrea_crd.NetworkPolicyPeer(
+            pod_selector = kubernetes.client.V1LabelSelector(),
+            namespace_selector = kubernetes.client.V1LabelSelector()
+        )
+    else:
+        np_name = generate_policy_name("recommend-reject-acnp")
+        ns, labels = applied_to.split(ROW_DELIMITER)
+        if ns in NAMESPACE_ALLOW_LIST:
+            return []
+        try:
+            labels_dict = json.loads(labels)
+        except Exception as e:
+            print("Error {}: labels {} in applied_to {} is not in json format".format(e, labels, applied_to))
+            return []
+        applied_to = antrea_crd.NetworkPolicyPeer(
+            pod_selector = kubernetes.client.V1LabelSelector(
+                match_labels = labels_dict
+            ),
+            namespace_selector = kubernetes.client.V1LabelSelector(
+                match_labels = {
+                    "kubernetes.io/metadata.name":ns
+                }
+            )
+        )
+    np = antrea_crd.ClusterNetworkPolicy(
+        kind = "ClusterNetworkPolicy",
+        api_version = "crd.antrea.io/v1alpha1",
+        metadata = kubernetes.client.V1ObjectMeta(
+            name = np_name,
+        ),
+        spec = antrea_crd.NetworkPolicySpec(
+            tier = "Baseline",
+            priority = DEFAULT_POLICY_PRIORITY,
+            applied_to = [applied_to],
+            egress = [antrea_crd.Rule(
+                action = "Reject",
+                to = [antrea_crd.NetworkPolicyPeer(
+                    pod_selector = kubernetes.client.V1LabelSelector())]
+            )],
+            ingress = [antrea_crd.Rule(
+                action = "Reject",
+                _from = [antrea_crd.NetworkPolicyPeer(
+                    pod_selector = kubernetes.client.V1LabelSelector())]
+            )],
+        )
+    )
+    return [dict_to_yaml(np.to_dict())]
+
+def recommend_k8s_policies(flows_df):
+    egress_rdd = flows_df.rdd.map(lambda flow:map_flow_to_egress(flow, k8s=True))\
+        .reduceByKey(lambda a, b: ("", a[1]+PEER_DELIMITER+b[1]))
+    ingress_rdd = flows_df.filter(flows_df.flowType != "pod_to_external")\
+        .rdd.map(map_flow_to_ingress)\
+        .reduceByKey(lambda a, b: (a[0]+PEER_DELIMITER+b[0], ""))
+    network_peers_rdd = ingress_rdd.union(egress_rdd)\
+                    .reduceByKey(combine_network_peers)
+    k8s_np_rdd = network_peers_rdd.flatMap(generate_k8s_np)
+    k8s_np_list = k8s_np_rdd.collect()
+    return k8s_np_list
+
+def recommend_antrea_policies(flows_df, option=1, deny_rules=True, to_services=True):
+    ingress_rdd = flows_df.filter(flows_df.flowType != "pod_to_external")\
+        .rdd.map(map_flow_to_ingress)\
+        .reduceByKey(lambda a, b: (a[0]+PEER_DELIMITER+b[0], ""))
```

Comment:
In PySpark, I think to achieve a 'namedTuple' like data structure I need to change the current RDD to the Dataframe type. This will involve lots of changes in the computation code. Could we mark this as a TODO for now?

Reply:
Surely it can be TODO, Ignore it for this PR.

The namedtuple would actually just be some syntactic sugar, where you access item in a tuple as if you were accessing an object.

I would not think this requires using a DataFrame, but if that's the case, it's surely not worth the effort.